lib/src/storage/backend/rocksdb.rs

```diff
@@ -261,19 +359,84 @@ impl Db {
 
             let mut cf_handles: Vec<*mut rocksdb_column_family_handle_t> =
                 vec![ptr::null_mut(); column_family_names.len()];
-            let db = ffi_result!(rocksdb_transactiondb_open_column_families_with_status(
-                options,
-                transactiondb_options,
-                c_path.as_ptr(),
-                c_column_families.len().try_into().unwrap(),
-                c_column_families
-                    .iter()
-                    .map(|cf| cf.as_ptr())
-                    .collect::<Vec<_>>()
-                    .as_ptr(),
-                cf_options.as_ptr() as *const *const rocksdb_options_t,
-                cf_handles.as_mut_ptr(),
-            ))
+
+            let name = c_path.as_ptr();
+            let num_column_families = c_column_families.len().try_into().unwrap();
+            let column_family_names = c_column_families
+                .iter()
+                .map(|cf| cf.as_ptr())
+                .collect::<Vec<_>>()
+                .as_ptr();
+            let column_family_options = cf_options.as_ptr() as *const *const rocksdb_options_t;
+            let column_family_handles = cf_handles.as_mut_ptr();
+
+            let db = match &open_mode {
```

Comment:
Nit: make `OpeningMode` copy with `#[derive(Clone, Clopy)]` to avoid having to make reference. 

We can also derive `Eq` to be able to make equality comparisons instead of using `if let`.

Reply:
i tried, but `PathBuf` does not implement `Copy`. Not sure how to continue

Reply:
That's a good point. I missed that. Sorry for the bad comment.